model/src/main/kotlin/PackageCuration.kt

```diff
@@ -86,8 +83,7 @@ data class PackageCuration(
                 return Semver.coerce(pkgId.version).satisfies(range)
             }
 
-            // `id.version` is invalid, but also not a version range. Therefore, coerce valid versions.
-            Semver.coerce(pkgId.version).satisfies(Semver.coerce(id.version).version)
+            return false
```

Comment:
I don't understand why we can keep on doing `coerce(pkgId.version)` in line 83, while `coerce(id.version)` is not okay anymore. Can you help?

Reply:
Thanks! I tried to find a case where the remaining `coerce()` can be problematic.
If we take above range `[1.0.0, 2.0.0]` and `pkg.id = "2.0.0.127"`, I believe `coerce("2.0.0.127")` yields `2.0.0`, so the range would be matched.

I wonder if it would work to solve the above mentioned use cases by doing the matching on a kind of "normalized" version string, whereas "normalize" would have effect only on version strings with less than 3 digits, like:
- `normalize("1") -> 1.0.0`
- `normalize("1.1) -> 1.1.0`

What do you think about that?

Reply:
Hm, even if this would make sense I believe it can be considered out of scope of this change / separate iteration. So, I'll just approve this one as-is.